### ISSUE TITLE
Problem with Detecting Quote Escape Sequences When Followed by Separator Character in Quote-Enclosed Fields

### ISSUE THREAD

**Issue by TheFanatr** [stevehansen/csv#27]
As can be seen below, when the read text contains a quote-enclosed field containing the quote escape sequence `""`, followed by the separator character `,`, the APIs for reading column values parse the data incorrectly.

![image](https://user-images.githubusercontent.com/16641853/67278051-114f2680-f47d-11e9-977f-6176536c3eef.png)

Also, as can be seen below, even when the `AllowSingleQuoteToEncloseFieldValues` flag is set, the incorrect parsing behaviour remains unchanged. I tested this flag in order to see if, for some reason, quote-enclosed fields were only detected with this mode activated, even though I know it is specifically to be used when fields are enclosed with the single quote character `'`.

![image](https://user-images.githubusercontent.com/16641853/67278230-6ee37300-f47d-11e9-914d-c770c5fd4f0b.png)

To summarize, it is valid for CSV data to contain fields enclosed by the `"` character, while also containing quote escape sequences such as `""` followed by separator characters such as the comma `,`, but this library does not seem to be able to parse this formatting correctly.

**Comment by pariesz** [comment]
This problem seem seems to have been introduced some version after 1.0.39, as it broke some of my tests after updating and I was forced to revert.

The following test demonstrates the problem:

```c#
[TestMethod]
public void TestCommaInQuotes() {
    var lines = CsvReader.ReadFromText("\"\"\"a,b\"\"\"", new CsvOptions { HeaderMode = HeaderMode.HeaderAbsent, Separator = ',' }).ToArray();
    Assert.AreEqual(1, lines.Length);
    Assert.AreEqual(1, lines[0].ColumnCount);
    Assert.AreEqual("\"a,b\"", lines[0][0]);
}
```

**Comment by stevehansen** [comment]
Issue #28 had the same problem, https://www.nuget.org/packages/Csv/1.0.55 fixes this now
